### PR TITLE
libgit2: update to 1.5.1

### PR DESCRIPTION
**Pull request by tranzystorek-io**
#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

File: srcpkgs/libgit2/patches/0001-clar-remove-ftrunacte-from-libgit2-tests.patch [hunk 1]
```diff
@@ -1,14 +1,5 @@
-From 12d73c418253a5c396465079c2808e07de17a1db Mon Sep 17 00:00:00 2001
-From: Peter Pettersson <boretrk@hotmail.com>
-Date: Thu, 14 Jul 2022 18:28:58 +0200
-Subject: [PATCH] clar: remove ftrunacte from libgit2 tests
-
----
- tests/libgit2/CMakeLists.txt | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
-
 diff --git a/tests/libgit2/CMakeLists.txt b/tests/libgit2/CMakeLists.txt
```
Comment (kruceter) [inline srcpkgs/libgit2/patches/0001-clar-remove-ftrunacte-from-libgit2-tests.patch:1]:
```suggestion
Backported from:
https://github.com/libgit2/libgit2/commit/12d73c418253a5c396465079c2808e07de17a1db

diff --git a/tests/libgit2/CMakeLists.txt b/tests/libgit2/CMakeLists.txt
```

I see nothing wrong in removing the commit header, but it is better to indicate that you backported the patch from the "main" branch to clear confusion.

File: srcpkgs/libgit2/patches/0001-clar-remove-ftrunacte-from-libgit2-tests.patch [hunk 2]
```diff
@@ -1,14 +1,5 @@
...
-index 27f421ad6..7f6fafe77 100644
+index 280b9a8..631fea8 100644
```
Comment (kruceter) [inline srcpkgs/libgit2/patches/0001-clar-remove-ftrunacte-from-libgit2-tests.patch:2]: Either remove them too or revert changes applied to the last selected line.

File: srcpkgs/libgit2/patches/0001-clar-remove-ftrunacte-from-libgit2-tests.patch (hunk 1 above)
Comment (tranzystorek-io) [reply srcpkgs/libgit2/patches/0001-clar-remove-ftrunacte-from-libgit2-tests.patch:4]: Done

**Comment by kruceter** [comment]
@Vaelatern, if you may?

**Review by Vaelatern — APPROVED** [review]

**Pull request merged**